Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -576,8 +560,21 @@ BytecodeGenerator::BytecodeGenerator(VM& vm, FunctionNode* functionNode, Unlinke
         // If we captured any formal parameter by name, then we use ScopedArguments. Otherwise we
         // use DirectArguments. With ScopedArguments, we lift all of our arguments into the
         // activation.
+        bool capturesAnyParameterByName = false;
+        if (functionNode->hasCapturedVariables()) {
+            for (size_t i = 0; i < parameters.size(); ++i) {
+                auto pattern = parameters.at(i).first;
+                ASSERT(pattern->isBindingNode());
```

Comment:
Why does `isSimpleParameterList` guarantee that `pattern->isBindingNode()` is true?

Reply:
It's guaranteed per implementation of `FunctionParameters::append()`:

```cpp
            bool isSimpleParameter = !hasDefaultParameterValue && pattern->isBindingNode();
            m_isSimpleParameterList &= isSimpleParameter;
```